gpu-enabler/src/main/scala/com/ibm/gpuenabler/CUDARDDUtils.scala

```diff
@@ -237,7 +240,78 @@ object CUDARDDImplicits {
     extends Serializable {
 
     def sc = rdd.sparkContext
+    
+    def autoCacheGPU(rdd: RDD[_]): Unit = {
+      val lineage = scala.collection.mutable.Set[RDD[_]]()
+      /**
+        * If calling this method in an action before the runJob,
+        * GPUEnabler cache RDDs on GPU automatically
+        * to remove redundant data transfers in a sequent of these
+        * every time GPUEnabler invoke actions has a sequent of GPU methods,
+        *
+        * gpuApiFlag shows the child type of a RDD.
+        * When the child RDD's API
+        *  case 0 => CPU API (ex, mapRDD)
+        *  case 1 => GPU API (don't divide a stage. ex, mapGPURDD)
+        *  case 2 => GPU API (divide a stage into two. ex,shuffleGPURDD)
+        */
+
+      if(judgeApiType(rdd) == 1) rdd.autoCacheGpu()
+      trackLineage(rdd, 1)
+
+      def trackLineage(rdd: RDD[_], gpuApiFlag: Int): Unit = {
+        val len = rdd.dependencies.length
+        len match {
+          case 0 => Unit
+          case 1 =>
+            val d = rdd.dependencies.head
+            val apiType = judgeApiType(d.rdd)
+            classifyDep(d.rdd, apiType, gpuApiFlag)
+          case _ =>
+            val frontDeps = rdd.dependencies.take(len - 1)
+            frontDeps.foreach( d => {
+              val apiType = judgeApiType(d.rdd)
+              classifyDep(d.rdd, apiType, gpuApiFlag)
+            })
+            val lastDep = rdd.dependencies.last
+            val apiType = judgeApiType(lastDep.rdd)
+            classifyDep(lastDep.rdd, apiType, gpuApiFlag)
+        }
+      }
+
+      def classifyDep(rdd: RDD[_], apiType: Int, gpuApiFlag: Int): Unit = {
```

Comment:
Would it be possible to define some constant literals for values of `gpuApiFlag` (i.e. `0`, `1`, `2`).
It is ease to understand source code.